### ISSUE TITLE
Column location Blocks off by one

### ISSUE THREAD

**Issue by nzakas** [csstree/csstree#296]
I've been investigating using CSS Tree in ESLint. First, thanks for the great project. This is exactly the level of detail we need.

As part of this investigation, I've discovered what I think is a bug. It appears that the column information for nodes is off by one.

## Reproduction Steps

**Version:** 3.0.0

Using this JS:

```js
import { parse, toPlainObject } from 'css-tree';

const root = parse('a { color: red }', { positions: true });
const plainRoot = toPlainObject(root);
```

The `loc` value for the `Block`node  is reported as:

```js
{
  source: '<unknown>',
  start: { offset: 3, line: 1, column: 4 },
  end: { offset: 15, line: 1, column: 16 }
}
```

Given that CSS Tree locations start at column 1, the correct start of the block as indicated by the `{` should be column 3.

I've created a StackBlitz that displays the problem here:
https://stackblitz.com/edit/stackblitz-starters-eoy8uv?file=index.js

**Regression?** This is different than the behavior in v2.3.1, which reports the location of `Block` in the same example as:

```js
{
  source: '<unknown>',
  start: { offset: 2, line: 1, column: 3 },
  end: { offset: 16, line: 1, column: 17 }
}
```

This is what I would expect, as I'd expect the opening and closing braces to be included in the location.

Is this intentional?

**Comment by lahmatiy** [comment]
@nzakas Yes, this is intentional. In CSSTree v3.0, the following changes were made:

- The `Block` no longer includes `{` and `}`.
- The `Atrule` and `Rule` now include `{` and `}` when they have a block.

In simpler terms, `{` and `}` are now part of the `Atrule` and `Rule`, not the `Block`. This change ensures that the responsibility for parsing and serializing braces is placed correctly: `Atrule` and `Rule` decide if they have a block based on encountering `{`, and are also responsible for serializing `{` and `}` in the CSS.

This update provides more flexibility, particularly for `Atrule`, which can now have a block as either `Block` or `StyleSheet` (which does not use braces). It also enables skipping block content parsing by replacing it with a `Raw` node if needed. Finally, this change in the `Block` node should eliminate the need for the `DeclarationList` node type, as the only distinguishing factor was the presence of braces.

Could you please tell why the block positions (or rather the curly braces) are important to you?

**Comment by nzakas** [comment]
That's interesting. I don't know enough about CSS parsing to have an informed opinion as to wheter `Atrule` and `Rule` should have this responsibility, I can only note that I've not seen this in ASTs of other languages that have blocks. Generally, I've seen curlies either represented as a `Block` node including the curlies (as in ESTree), or there is no formal block node, as it is just considered a container around a list of values (as in JSON and GraphQL).

To give you a concrete example of how this caused some confusion: I'm working on a `no-empty-block` rule, where a block without children is flagged. The location of the node is then passed into ESLint so the text can be highlighted in an editor or on the command line. If I pass `Block.loc`, the highlight will just be inside the curlies, which is problematic for `{}`, in which case, nothing will be highlighted. Looking at this from a user perspective, I think they'd expect the curlies to be highlighted.

That said, I can always expand the range of the location by one in each direction (assuming you are always starting the block at immediate next character and the ending at the immediate previous character), but that also means any rule that works with blocks will have to do the same thing. So, I'll probably end up just modifying the `loc` on each `Block` to include the curly braces and on each `Atrule` and `Rule` to exclude them.

**Comment by lahmatiy** [comment]
I don't have a strong position here. As mentioned earlier, I initially considered moving the responsibility for handling curly braces to `Rule` and `Atrule` nodes, expecting it would allow for varied node types within `block`. However, with recent CSS updates (nesting and friends), it has become complex from a parser's standpoint to differentiate between `Block` and `StyleSheet` for nested blocks. Therefore, I decided to treat everything as a `Block`, as differentiating no longer makes much sense.

From a user’s perspective, it’s reasonable that curly braces should belong to a `Block`. Based on this reasoning, my initial decision now seems mistaken. I consider this a bug and will be rolling back the changes. Fortunately, this primarily affects the location of `Block` nodes, so the impact is minimal.
